Signal/src/ViewControllers/MediaMessageView.swift

```diff
@@ -365,4 +390,51 @@ class MediaMessageView: UIView, OWSAudioAttachmentPlayerDelegate {
         audioPlayButton?.setImage(image, for: .normal)
         audioPlayButton?.imageView?.tintColor = UIColor.ows_materialBlue()
     }
+
+    // MARK: - Video Playback
+
+    func videoTapped(sender: UIGestureRecognizer) {
+        guard let dataUrl = attachment.dataUrl else {
+            return
+        }
+        guard sender.state == .recognized else {
+            return
+        }
+        guard let videoPlayer = MPMoviePlayerController(contentURL: dataUrl) else {
+            return
+        }
+        videoPlayer.prepareToPlay()
+
+        NotificationCenter.default.addObserver(forName: .MPMoviePlayerWillExitFullscreen, object: nil, queue: nil) { [weak self] _ in
+            self?.moviePlayerWillExitFullscreen()
+        }
+        NotificationCenter.default.addObserver(forName: .MPMoviePlayerDidExitFullscreen, object: nil, queue: nil) { [weak self] _ in
+            self?.moviePlayerDidExitFullscreen()
+        }
+
+        videoPlayer.controlStyle = .default
+        videoPlayer.shouldAutoplay = true
+
+        self.addSubview(videoPlayer.view)
+        videoPlayer.view.frame = self.bounds
+        self.videoPlayer = videoPlayer
+        videoPlayer.view.autoPinToSuperviewEdges()
+        ViewControllerUtils.setAudioIgnoresHardwareMuteSwitch(true)
+        videoPlayer.setFullscreen(true, animated:false)
+    }
+
+    private func moviePlayerWillExitFullscreen() {
+        clearVideoPlayer()
+    }
+
+    private func moviePlayerDidExitFullscreen() {
+        clearVideoPlayer()
+    }
+
+    private func clearVideoPlayer() {
+        videoPlayer?.stop()
+        videoPlayer?.view.removeFromSuperview()
+        videoPlayer = nil
+        ViewControllerUtils.setAudioIgnoresHardwareMuteSwitch(false)
```

Comment:
I forgot what this method did and just took a look. It strikes me as weird that we'd be requesting a "record" session at this place. I know this is existing behavior, but I've made a card to audit the usage of this method.

Reply:
Good call.  I can't recall if this is a mistake or not.